UnityProject/Assets/Scripts/3D/ConvertTo3D.cs

```diff
@@ -1,15 +1,34 @@
+using System;
 using UnityEngine;
 using UnityEngine.Rendering;
 
 namespace _3D
 {
 	public class ConvertTo3D : MonoBehaviour
 	{
+		private RegisterTile RegisterTile;
+
+		public void Awake()
+		{
+			RegisterTile = this.GetComponent<RegisterTile>();
```

Comment:
```cs
RegisterTile ??= GetComponent<RegisterTile>();
```

Reply:
like, technically it is unneeded null check does not Set anywhere, 
we need a agreement on this where components should 
A grab everything it needs on awake
B have everything Populated in the editor
which one is faster?